src/libraries/Microsoft.PowerFx.Core/Binding/INameResolver.cs

```diff
@@ -36,6 +36,8 @@ internal interface INameResolver
 
         IEnumerable<TexlFunction> Functions { get; }
 
+        IDictionary<string, NameLookupInfo> VariableNames { get; }
```

Comment:
>IDictionary [](http://example.com/codeflow?start=8&length=11)

let's chat about this. 

At the very least, it should be a ReadOnlyDictionary (since caller can't mutate it)